### ISSUE TITLE
Need to make the Overworld Website Mobile-Friendly

### ISSUE THREAD

**Issue by PathakPratik** [danielgrijalva/overworld#178]
When we open the Overworld website on a mobile, we see that it is not responsive and hence very difficult to use. 

We could use Flexbox and Media queries or a framework like Bootstrap for achieving a fully responsive website.

A screenshot showing how the home page is rendered in mobile is attached below:-

![Overworld](https://user-images.githubusercontent.com/17859354/71318924-5adbe200-24bd-11ea-9eff-b95a5b320a4f.png)

**Comment by danielgrijalva** [comment]
Definitely, this is something that’s been bugging me since the start and I’m just delaying it.
I thought that using the [grid system](https://react.semantic-ui.com/collections/grid/) from Semantic UI would cover this but apparently nope (or maybe we are missing something and need to implement them correctly…?)

I think that one of the main problems is that I set a `min-width` for the body in index.css

https://github.com/danielgrijalva/overworld/blob/6b17acaf2e366e33c64f8f5ca65b938a4f2d97f3/frontend/src/index.css#L11
===
I’ve been playing with flexbox and CSS grid myself so I’ll see if I can work on this in my free time. 
In the meantime I’ll share this issue with everyone, so if someone’s reading this feel free to contribute! ✨

**Comment by tlaudahl** [comment]
I’ll look into this as well. I’m pretty comfortable with flex box and media queries. Would there be any objection to converting the css to sass/less? @danielgrijalva  It makes writing css way easier, especially media queries in my opinion. If not that’s fine

**Comment by PathakPratik** [comment]
Even I think using Sass/Less would be great.

**Comment by danielgrijalva** [comment]
@tlaudahl go ahead. I was thinking of using styled components but sass seems more flexible 😃

**Comment by tlaudahl** [comment]
I’ve been busy with Christmas and New Years stuff going on but I got time now I’ll finally start on this when I get home from work

**Comment by PathakPratik** [comment]
Had a look at the code for Home page. Removing the min-width set in index.css does help. After which some fixes specific for each component are required. The main task though looks like making the navbar responsive. We may have to create a hamburger menu for mobile devices. What do you guys think?

**Comment by PathakPratik** [comment]
Also, opened a PR which makes the settings page responsive. Just adding 'stackable' property to grid did the trick. #186

**Comment by tlaudahl** [comment]
Probably a good idea, still think it's a good idea to convert all the css to sass? It will get processed into css again anyway but I'll have to go through every css file and make a new sass file for it and compile it

**Comment by PathakPratik** [comment]
How about using scss? 
http://thesassway.com/editorial/sass-vs-scss-which-syntax-is-better

It could help us incrementally use sass features with minimal efforts.

**Comment by danielgrijalva** [comment]
SCSS seems better to me since more people could jump into it from CSS without knowing much about SASS. 

And the navbar sure is the main problem to tackle—hamburger menu sound great but couldn’t find anything related in Semantic UI docs.

**Comment by Ofineo** [comment]
Has this progresses since the last post? I would like to help if it's still needed.

**Comment by danielgrijalva** [comment]
@Ofineo @PathakPratik @tlaudahl hey guys, sorry for being absent. I just fixed a conflict in PR #191, which introduces SCSS. I merged the branch so now we got SCSS fully working :)

**Comment by Panda4817** [comment]
Hello, 

I think this project is a great idea and I had some ideas about how to make the site more mobile responsive. Maybe a stackable navbar would be a good idea. I have forked it and got it working locally. I have tinkered with the scss and this is the result:

![homepage](https://i.imgur.com/iw9bOGv.gif)

![logged in](https://i.imgur.com/JlNDp7F.gif)

These are files that have been changed (only scss, added a stackable attritube and added a class name). 
src/index.scss
 -  added a bigger margin to certain containers that are affected by stacked menu on mobile screens <code> 
&.container:not(.backdrop):not(.profile-menu-wrapper):not(.footer) {
    @media (max-width: 767px) {
      margin-top: 180px;
    }
  }
</code>

src/modules/app/components/backdrop/styles.scss
-  added no padding-bottom when menu is stacked (I think it looks better on mobile  screens) <code>
.backdrop {
  padding-bottom: 400px;
  @media (max-width: 767px) {
    padding-bottom: 0px;
  }
}
</code>

src/modules/app/components/footer/index.js
- added footer class name to footer container component so it can be easily excluded from thee extra margin added to containers<code> Container className="footer"  </code>

src/modules/app/components/navbar/index.js
-  added stackable attribute to menu component<code>
Menu className="navbar" inverted secondary stackable
</code>

src/modules/app/components/navbar/styles.scss
-  Added the following css to the navbar class<code>
.navbar {
  @media (min-width: 1199px) {
    max-width: 1200px;
  }
.grid {
    @media (max-width: 800px) {
      margin-left: -40px !important;
    }
    @media (max-width: 767px) {
      margin-left: -14px !important;
    }
  }
}
</code>

I can create a pull request if this is all acceptable or I am open to any suggestions.

**Comment by danielgrijalva** [comment]
Hey @Panda4817 thanks a lot! this is a nice improvement.

> Maybe a stackable navbar would be a good idea

Can you work on the stackable navbar?

**Comment by JohnMachado11** [comment]
I have experience with Bootstrap, HTML, CSS and Javascript. My last project was 100% mobile and desktop friendly. Any progress been made on the site conversion to make it mobile friendly?

**Comment by Panda4817** [comment]
@danielgrijalva I have done a PR, please take a look 🙂

**Comment by danielgrijalva** [comment]
@Panda4817 oh yeah sorry. I've merged it now, thanks! ❤️

**Comment by ItzaMi** [comment]
Is this issued finished?

**Comment by HirenChafekar** [comment]
Is the issue still open?

**Comment by VhugoJc** [comment]
Is it open?
